### PR TITLE
doc: Glossary Update

### PR DESCRIPTION
**Pull request by FrancescoSer**
Updated the Glossary.

Signed-off-by: Francesco Domenico Servidio <francesco.servidio@nordicsemi.no>

### CODE REVIEW

**Review by wiba-nordic — APPROVED** [review]
Looks good.

File: doc/nrf/glossary.rst [hunk 1]
```diff
@@ -414,11 +457,17 @@ Glossary
...
+   Received Signal Strength Indication (RSSI)
+      An indication of the power of a received radio signal.
+
    Repository
       The entire set of files and folders of which a project is composed, together with the revision history of each file.
       It is often composed of multiple branches.
       It is also known as *Git repository* or *Git project*, and is similar to a *Git submodule*.
 
+   Request to Send (RTS)
+      In flow control, the transmitting end is ready and requesting the far end for a permission to transfer data.
```
Comment (greg-fer) [inline doc/nrf/glossary.rst:469]: See earlier comments.

File: doc/nrf/glossary.rst [hunk 2]
```diff
@@ -61,6 +67,9 @@ Glossary
       A secure system depends on building and maintaining a chain of trust through all the layers in the system.
       Each step in this chain guarantees that the next step can be trusted to have certain properties because any unauthorized modification of a subsequent step will be detected and the process halted.
 
+   Clear to Send (CTS)
+      In flow control, the receiving end is ready and telling the far end to start sending.
```
[the remark this answers is not in the document]
Comment (FrancescoSer) [reply doc/nrf/glossary.rst]: fixed

File: doc/nrf/glossary.rst [hunk 3]
```diff
@@ -85,6 +94,9 @@ Glossary
    Cortex Microcontroller Software Interface Standard (CMSIS)
       A vendor-independent hardware abstraction layer for the Cortex-M processor series that defines generic tool interfaces.
 
+   Data Terminal Ready (DTR)
+      A control signal in RS-232 serial communications transmitted from data terminal equipment, such as a computer, to data communications equipment.
```
[the remark this answers is not in the document]
Comment (FrancescoSer) [reply doc/nrf/glossary.rst]: fixed

File: doc/nrf/glossary.rst [hunk 4]
```diff
@@ -181,6 +203,9 @@ Glossary
    Integrated Development Environment (IDE)
       A software application that provides facilities for software development.
 
+   International Mobile (Station) Equipment Identity (IMEI)
+      A unique code consisting of 14 digits and a check digit for identifying 3GPP-based mobile devices.
```
[the remark this answers is not in the document]
Comment (FrancescoSer) [reply doc/nrf/glossary.rst]: right, fixed

File: doc/nrf/glossary.rst (hunk 1 above)
Comment (FrancescoSer) [reply doc/nrf/glossary.rst]: fixed

File: doc/nrf/glossary.rst [hunk 5]
```diff
@@ -321,6 +347,9 @@ Glossary
    Noise Factor (NF)
       The relation of the Signal-to-Noise Ratio (SNR) in the device input to the SNR in the device output.
 
+   Non-volatile Memory (NVM)
+      Memory that can retrieve stored information even after having been power-cycled.
```
[the remark this answers is not in the document]
Comment (FrancescoSer) [reply doc/nrf/glossary.rst]: fixed

**Comment by FrancescoSer** [comment]
@carlescufi @mbolivar-nordic Should the definition of HWFC and of UART HWFC be merged in one?

File: doc/nrf/glossary.rst [hunk 6]
```diff
@@ -414,21 +457,27 @@ Glossary
...
+
    Repository
       The entire set of files and folders of which a project is composed, together with the revision history of each file.
       It is often composed of multiple branches.
       It is also known as *Git repository* or *Git project*, and is similar to a *Git submodule*.
 
+   Request to Send (RTS)
+      In :term:`flow control <Hardware Flow Control (HWFC)>`, the transmitting end is ready and requesting the far end for permission to transfer data.
+
    Root of Trust (RoT)
```
Comment (peknis) [inline doc/nrf/glossary.rst:471]: Maybe we need to add RPC (remote procedure call)

Comment (FrancescoSer) [reply doc/nrf/glossary.rst:471]: @peknis Do you also have a clear definition for it you would like to add?

Comment (peknis) [reply doc/nrf/glossary.rst:471]: Did not find in our docs, but Google gave for example this: https://www.w3.org/History/1992/nfs_dxcern_mirror/rpc/doc/Introduction/WhatIs.html

Comment (FrancescoSer) [reply doc/nrf/glossary.rst:474]: added, will tag reviewers on that one

File: doc/nrf/glossary.rst [hunk 7]
```diff
@@ -414,21 +457,30 @@ Glossary
    Receive Data (RXD)
       A signal line in a serial interface that receives data from another device.
 
+   Received Signal Strength Indication (RSSI)
+      An indication of the power of a received radio signal.
+
+   Remote Procedure Call (RPC)
+      A form of inter-process communication that allows for calling a function on remote processors from a local processor, both synchronously and asynchronously.
```
Comment (FrancescoSer) [inline doc/nrf/glossary.rst:464]: @carlescufi @mbolivar-nordic Can you check if this definition is actually correct in the NCS context?

File: doc/nrf/glossary.rst [hunk 8]
```diff
@@ -166,8 +184,12 @@ Glossary
    Global Positioning System (GPS)
       A satellite-based radio navigation system that provides its users with accurate location and time information over the globe.
 
+   Hardware Flow Control (HWFC)
+      A handshaking mechanism used to prevent an overflow of bytes in modems.
+      It utilizes two dedicated pins on the RS-232 connector, *Request to Send* and *Clear to Send*.
```
Comment (peknis) [inline doc/nrf/glossary.rst:189]: Maybe uses instead of utilizes?

Comment (FrancescoSer) [reply doc/nrf/glossary.rst]: fixed

**Review by peknis — APPROVED** [review]

**Comment by nvlsianpu** [comment]
@FrancescoSer Can you rebase?

**Pull request merged**